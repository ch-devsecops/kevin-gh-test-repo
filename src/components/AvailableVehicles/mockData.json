{
  "uid": "1ebf252b-374c-4c65-a977-5a1bf68b2031",
  "componentName": "AvailableVehicles",
  "dataSource": "{1E951413-E986-475B-BA74-D7BA568D167D}",
  "params": {},
  "fields": {
    "data": {
      "value": {
        "anchorId": {},
        "gtmTitle": {},
        "gtmCategory": {},
        "gtmModelName": {},
        "gtmTrimName": {},
        "gtmBodyStyle": {
          "value": "GTM Body Style"
        },
        "gtmInteractionType": {},
        "catalog": {
          "fields": {
            "name": "Honda",
            "models": [
              {
                "items": [
                  {
                    "name": "ODYSSEY",
                    "modelYears": [
                      {
                        "name": "2023",
                        "trims": [
                          {
                            "name": "EX-L",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10797-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10797"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "price": {
                                      "value": "36,000.00"
                                    },
                                    "trim": {
                                      "value": "EX-L"
                                    },
                                    "tooltipText": {
                                      "value": "This is a tooltip"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/grey-crv.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "Grey"
                                    }
                                  }
                                ]
                              }
                            ]
                          },
                          {
                            "name": "TOURING",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10798-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10798"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "price": {
                                      "value": "42,000.00"
                                    },
                                    "trim": {
                                      "value": "Touring"
                                    },
                                    "tooltipText": {
                                      "value": "This is Touring"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/blue-accord.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "blue"
                                    }
                                  }
                                ]
                              }
                            ]
                          },
                          {
                            "name": "BLACK EDITION",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10799-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10799"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "exteriorColorKey": {
                                      "value": "orange"
                                    },
                                    "price": {
                                      "value": "57,550.00"
                                    },
                                    "trim": {
                                      "value": "black-edition"
                                    },
                                    "tooltipText": {
                                      "value": "This is Black Edition"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/grey-civic.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "blue"
                                    }
                                  }
                                ]
                              }
                            ]
                          },
                          {
                            "name": "EX",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10796-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10796"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "price": {
                                      "value": ""
                                    },
                                    "trim": {
                                      "value": "AWD"
                                    },
                                    "tooltipText": {
                                      "value": "This is EX model"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/orange-civic.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "blue"
                                    }
                                  }
                                ]
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  }
                ]
              }
            ]
          }
        }
      }
    },
    "gtmCategory": {
      "id": "86f35331-a03f-4d3e-b8d2-6142e38ad8f3",
      "url": "/sitecore/System/Foundation/common/GTM-Categories/Others",
      "name": "Others",
      "displayName": "Others",
      "fields": {
        "value": {
          "value": "others"
        }
      }
    },
    "gtmTitle": {
      "value": "GTM Title "
    },
    "gtmBodyStyle": {
      "value": "GTM Body Style"
    },
    "gtmInteractionType": {
      "value": "GTM Interaction Type"
    },
    "gtmTrimName": {
      "value": "GTM Trim Name"
    },
    "gtmModelName": {
      "value": "GTM Model Name"
    }
  }
}
