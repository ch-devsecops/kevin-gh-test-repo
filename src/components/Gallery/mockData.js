export const strings = {
  Shared: {
    Gallery: {
      galleryModalAria: 'gallery modal',
      exitModalAria: 'exit modal',
      fullscreenModeAria: 'view full screen mode',
      nextImageAria: 'view next image',
      prevImageAria: 'view previous image',
      nextThumbnailPageAria: 'view next thumbnail page',
      prevThumbnailPageAria: 'view previous thumbnail page',
      hideAria: 'hide content',
      hideLabel: 'Hide',
    },
  },
};

export default { strings };
