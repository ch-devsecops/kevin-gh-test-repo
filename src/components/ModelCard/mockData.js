// eslint-disable-next-line import/prefer-default-export
export const ModelCardData = {
  Model: {
    Name: 'Model_Name',
    NameBadge: {
      src: '/data/media/img/model-card/ilx_badge.svg',
      alt: 'model name badge alt text',
    },
    ModelKey: 'mdx',
    modelYear: 2022,
  },
  Trims: {
    SpecialTrimType: 'Available',
    TaglineOverride: 'Performance Luxury Sedan',
    NameBadge: {
      src: '/data/media/img/model-card/type_s.svg',
      alt: 'trim name badge alt text',
    },
  },
  Transmission: {
    IsBuildable: true,
    Name: 'Transmission A',
    TransmissionModelCode: '10201-Automatic',
    HasSpecs: true,
    InteriorColorKey: 'kkebony_leatherette_^my20_ilx',
    ExteriorColorKey: 'bkebony_leatherette_^my20_rdx',
    Trim: 'rdx_10226',
  },
};
