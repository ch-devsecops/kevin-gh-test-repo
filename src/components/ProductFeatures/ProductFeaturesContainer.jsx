import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Box, Accordion } from '@honda-canada/design-system-react';

import NestedAccordionTitle from '../../utils/components/NestedAccordionTitle';

import themeStyles from './ProductFeatures.styles';
import FeaturesList from './FeaturesList';

const Container = themeStyles.apply(Box, 'Container');
const FeatureWrapper = themeStyles.apply(Box, 'FeatureWrapper');

const ProductFeaturesContainer = ({ features, gtmTags }) => {
  const [isAccordion, setIsAccordion] = useState(false);

  if (!isAccordion && features?.[0]?.accordionTitle) {
    setIsAccordion(true);
  }

  const content = features?.map((feature, i) => {
    if (feature.accordionTitle) {
      return {
        ...feature,
        title: <NestedAccordionTitle>{feature.title}</NestedAccordionTitle>,
        content: (
          <FeatureWrapper isAccordion={isAccordion}>
            {feature.content.map((item, j) => (
              <FeaturesList key={`${item.title}_${j}`} feature={item} />
            ))}
          </FeatureWrapper>
        ),
      };
    }
    return <FeaturesList key={`${feature.title}_${i}`} feature={feature} />;
  });

  return (
    <Container data-gtm-component-type={gtmTags?.type} tabIndex={0} data-testid="product-features-content">
      {isAccordion ? (
        <Accordion
          items={content}
          iconPosition="left"
          iconName="animatedArrowRightDown"
          iconColorOverride="primary"
          panelBackgroundColor="grey.3"
          borderColor="transparent"
          isCompact
          contentContainerStyles={{ padding: 0, marginBottom: ['xs', 's'] }}
        />
      ) : (
        content
      )}
    </Container>
  );
};

ProductFeaturesContainer.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          text: PropTypes.string,
          tooltip: PropTypes.string,
        }),
      ),
    }),
  ),
  gtmTags: PropTypes.shape({
    type: PropTypes.string,
  }),
};

export default ProductFeaturesContainer;
