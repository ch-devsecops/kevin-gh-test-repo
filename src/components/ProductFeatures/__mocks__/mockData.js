/* eslint-disable quotes */
export const inputString =
  '<Four-Stroke Engine Design> \r\nWorld-renowned Honda quality combines proven reliability with no oil-gasoline mixing.\r\n<Centrifugal Clutch> \r\nHonda\'s centrifugal clutch facilitates the running of the engine at idle, with no propeller rotation, for tight navigation or just idling at the dock.\r\n<Lightweight, Compact Design> \r\nEasily transportable, this little outboard is ideal for small boats, dinghies and canoes. Lightweight 4-stroke design and is the only 4-stroke in the 2.3-hp class.\r\n<Transistorized Ignition> \r\nMaintenance-free, high-energy output for quick and easy starting.\r\n<Forced-Air Cooled Engine>\r\nNo water pump to worry about being plugged with weeds or damaged by sand, and less weight to carry.\r\n<Emergency Stop Switch and Lanyard>\r\nStandard safety feature allows the operator to clip the safety switch lanyard or tether to his or her wrist, which will automatically activate the engine stop switch if the operator leaves the helm.\r\n<Shallow-Water Drive> \r\nReduces the risk of engine damage while navigating in shallow water.\r\n<Sacrificial Anode> \r\nA sacrificial anode mounted on the lower unit protects other metal components from corrosion.\r\n<Three-Year Warranty> \r\nThe best outboards in the industry are backed by the best warranty in the business, 3 years parts and labour.\r\n<Two Models to Choose From>\r\nThe BF2.3DK2SCHC and BF2.3DK2LCHC models have a centrifugal drive clutch, which offers a neutral-gear position at idle. Both models have a tiller-mounted twist grip throttle/360° swivel steering.\r\n<360º Degree Swivel Steering> \r\nProvides excellent manoeuvrability in tight areas.\r\n<Integral Fuel Tank>\r\nFacilitates easy removal and portability as one unit. Vent may be closed for transport or extended storage periods.\r\n<Carburetor Drain Screw> \r\nAllows quick and easy draining of the carburetor\'s float bowl for transport or extended storage periods.\r\n<Twist Grip Throttle with Friction Adjuster>\r\nAllows for superior throttle control, and the throttle tensioner minimizes driver fatigue. \r\n<Large Front-Mounted Carrying Handle>\r\nProvides superior portability and ease of installation.\r\n<Fold-Down Tiller Handle> \r\nA more compact design allows for easy transportation and storage.\r\n<Available In Two Shaft Lengths> \r\nStandard 15" or 20" shaft lengths allow for a wider range of boat applications';

export const inputStringWithAccordion =
  '<<Performance and Efficiency>>\r\n<Four-Stroke Engine Design> \r\nWorld-renowned Honda quality combines proven reliability with no oil-gasoline mixing.\r\n<Centrifugal Clutch> \r\nHonda\'s centrifugal clutch facilitates the running of the engine at idle, with no propeller rotation, for tight navigation or just idling at the dock.\r\n<Lightweight, Compact Design> \r\nEasily transportable, this little outboard is ideal for small boats, dinghies and canoes. Lightweight 4-stroke design and is the only 4-stroke in the 2.3-hp class.\r\n<Transistorized Ignition> \r\nMaintenance-free, high-energy output for quick and easy starting.\r\n<<Reliability and Durability>>\r\n<Forced-Air Cooled Engine>\r\nNo water pump to worry about being plugged with weeds or damaged by sand, and less weight to carry.\r\n<Emergency Stop Switch and Lanyard>\r\nStandard safety feature allows the operator to clip the safety switch lanyard or tether to his or her wrist, which will automatically activate the engine stop switch if the operator leaves the helm.\r\n<Shallow-Water Drive> \r\nReduces the risk of engine damage while navigating in shallow water.\r\n<Sacrificial Anode> \r\nA sacrificial anode mounted on the lower unit protects other metal components from corrosion.\r\n<Three-Year Warranty> \r\nThe best outboards in the industry are backed by the best warranty in the business, 3 years parts and labour.\r\n<<Convenience and Ease of Use>>\r\n<Two Models to Choose From>\r\nThe BF2.3DK2SCHC and BF2.3DK2LCHC models have a centrifugal drive clutch, which offers a neutral-gear position at idle. Both models have a tiller-mounted twist grip throttle/360° swivel steering.\r\n<360º Degree Swivel Steering> \r\nProvides excellent manoeuvrability in tight areas.\r\n<Integral Fuel Tank>\r\nFacilitates easy removal and portability as one unit. Vent may be closed for transport or extended storage periods.\r\n<Carburetor Drain Screw> \r\nAllows quick and easy draining of the carburetor\'s float bowl for transport or extended storage periods.\r\n<Twist Grip Throttle with Friction Adjuster>\r\nAllows for superior throttle control, and the throttle tensioner minimizes driver fatigue. \r\n<Large Front-Mounted Carrying Handle>\r\nProvides superior portability and ease of installation.\r\n<Fold-Down Tiller Handle> \r\nA more compact design allows for easy transportation and storage.\r\n<Available In Two Shaft Lengths> \r\nStandard 15" or 20" shaft lengths allow for a wider range of boat applications';

export const inputRequest = hasAccordion => ({
  Url: 'BF2_3',
  AutoDataCode: '',
  Description:
    "The world's lightest 4-stroke outboard, it’s ideal for tenders, small fishing boats or as auxiliary power.",
  Disclaimer: '',
  SubTitle: '',
  TextHighlights: [],
  TextFeatures: hasAccordion ? inputStringWithAccordion : inputString,
  IconAssetUrl: null,
  Id: 12034,
  Key: 'bf2',
  Name: 'BF2.3',
  GtmName: 'BF2.3',
  SeatsCount: 0,
  Features: [],
  Highlights: [],
  IsFutureVehicle: false,
  Co2Badge: {
    IsModelBadge: false,
    BadgeTargetUrl: '',
    BadgeAssetPath: null,
    Co2Emission: 0,
    Co2SegmentAverage: 0,
    TrimName: 'BF2.3',
    TrimGtmName: 'BF2.3',
  },
  DisplayOrder: 1,
  FreightPdiCost: 158.0,
  DisplayName: null,
});

export const outputObjectWithAccordion = [
  {
    accordionTitle: 'Performance and Efficiency',
    title: 'Performance and Efficiency',
    key: 'Performance and Efficiency',
    content: [
      {
        list: [{ text: 'World-renowned Honda quality combines proven reliability with no oil-gasoline mixing.' }],
        title: 'Four-Stroke Engine Design',
      },
      {
        list: [
          {
            text: "Honda's centrifugal clutch facilitates the running of the engine at idle, with no propeller rotation, for tight navigation or just idling at the dock.",
          },
        ],
        title: 'Centrifugal Clutch',
      },
      {
        list: [
          {
            text: 'Easily transportable, this little outboard is ideal for small boats, dinghies and canoes. Lightweight 4-stroke design and is the only 4-stroke in the 2.3-hp class.',
          },
        ],
        title: 'Lightweight, Compact Design',
      },
      {
        list: [{ text: 'Maintenance-free, high-energy output for quick and easy starting.' }],
        title: 'Transistorized Ignition',
      },
    ],
  },
  {
    accordionTitle: 'Reliability and Durability',
    title: 'Reliability and Durability',
    key: 'Reliability and Durability',
    content: [
      {
        list: [
          {
            text: 'No water pump to worry about being plugged with weeds or damaged by sand, and less weight to carry.',
          },
        ],
        title: 'Forced-Air Cooled Engine',
      },
      {
        list: [
          {
            text: 'Standard safety feature allows the operator to clip the safety switch lanyard or tether to his or her wrist, which will automatically activate the engine stop switch if the operator leaves the helm.',
          },
        ],
        title: 'Emergency Stop Switch and Lanyard',
      },
      {
        list: [{ text: 'Reduces the risk of engine damage while navigating in shallow water.' }],
        title: 'Shallow-Water Drive',
      },
      {
        list: [
          { text: 'A sacrificial anode mounted on the lower unit protects other metal components from corrosion.' },
        ],
        title: 'Sacrificial Anode',
      },
      {
        list: [
          {
            text: 'The best outboards in the industry are backed by the best warranty in the business, 3 years parts and labour.',
          },
        ],
        title: 'Three-Year Warranty',
      },
    ],
  },
  {
    accordionTitle: 'Convenience and Ease of Use',
    title: 'Convenience and Ease of Use',
    key: 'Convenience and Ease of Use',
    content: [
      {
        list: [
          {
            text: 'The BF2.3DK2SCHC and BF2.3DK2LCHC models have a centrifugal drive clutch, which offers a neutral-gear position at idle. Both models have a tiller-mounted twist grip throttle/360° swivel steering.',
          },
        ],
        title: 'Two Models to Choose From',
      },
      { list: [{ text: 'Provides excellent manoeuvrability in tight areas.' }], title: '360º Degree Swivel Steering' },
      {
        list: [
          {
            text: 'Facilitates easy removal and portability as one unit. Vent may be closed for transport or extended storage periods.',
          },
        ],
        title: 'Integral Fuel Tank',
      },
      {
        list: [
          {
            text: "Allows quick and easy draining of the carburetor's float bowl for transport or extended storage periods.",
          },
        ],
        title: 'Carburetor Drain Screw',
      },
      {
        list: [{ text: 'Allows for superior throttle control, and the throttle tensioner minimizes driver fatigue. ' }],
        title: 'Twist Grip Throttle with Friction Adjuster',
      },
      {
        list: [{ text: 'Provides superior portability and ease of installation.' }],
        title: 'Large Front-Mounted Carrying Handle',
      },
      {
        list: [{ text: 'A more compact design allows for easy transportation and storage.' }],
        title: 'Fold-Down Tiller Handle',
      },
      {
        list: [{ text: 'Standard 15" or 20" shaft lengths allow for a wider range of boat applications' }],
        title: 'Available In Two Shaft Lengths',
      },
    ],
  },
];

export const outputObject = [
  {
    list: [{ text: 'World-renowned Honda quality combines proven reliability with no oil-gasoline mixing.' }],
    title: 'Four-Stroke Engine Design',
  },
  {
    list: [
      {
        text: "Honda's centrifugal clutch facilitates the running of the engine at idle, with no propeller rotation, for tight navigation or just idling at the dock.",
      },
    ],
    title: 'Centrifugal Clutch',
  },
  {
    list: [
      {
        text: 'Easily transportable, this little outboard is ideal for small boats, dinghies and canoes. Lightweight 4-stroke design and is the only 4-stroke in the 2.3-hp class.',
      },
    ],
    title: 'Lightweight, Compact Design',
  },
  {
    list: [{ text: 'Maintenance-free, high-energy output for quick and easy starting.' }],
    title: 'Transistorized Ignition',
  },
  {
    list: [
      {
        text: 'No water pump to worry about being plugged with weeds or damaged by sand, and less weight to carry.',
      },
    ],
    title: 'Forced-Air Cooled Engine',
  },
  {
    list: [
      {
        text: 'Standard safety feature allows the operator to clip the safety switch lanyard or tether to his or her wrist, which will automatically activate the engine stop switch if the operator leaves the helm.',
      },
    ],
    title: 'Emergency Stop Switch and Lanyard',
  },
  {
    list: [{ text: 'Reduces the risk of engine damage while navigating in shallow water.' }],
    title: 'Shallow-Water Drive',
  },
  {
    list: [{ text: 'A sacrificial anode mounted on the lower unit protects other metal components from corrosion.' }],
    title: 'Sacrificial Anode',
  },
  {
    list: [
      {
        text: 'The best outboards in the industry are backed by the best warranty in the business, 3 years parts and labour.',
      },
    ],
    title: 'Three-Year Warranty',
  },
  {
    list: [
      {
        text: 'The BF2.3DK2SCHC and BF2.3DK2LCHC models have a centrifugal drive clutch, which offers a neutral-gear position at idle. Both models have a tiller-mounted twist grip throttle/360° swivel steering.',
      },
    ],
    title: 'Two Models to Choose From',
  },
  { list: [{ text: 'Provides excellent manoeuvrability in tight areas.' }], title: '360º Degree Swivel Steering' },
  {
    list: [
      {
        text: 'Facilitates easy removal and portability as one unit. Vent may be closed for transport or extended storage periods.',
      },
    ],
    title: 'Integral Fuel Tank',
  },
  {
    list: [
      {
        text: "Allows quick and easy draining of the carburetor's float bowl for transport or extended storage periods.",
      },
    ],
    title: 'Carburetor Drain Screw',
  },
  {
    list: [{ text: 'Allows for superior throttle control, and the throttle tensioner minimizes driver fatigue. ' }],
    title: 'Twist Grip Throttle with Friction Adjuster',
  },
  {
    list: [{ text: 'Provides superior portability and ease of installation.' }],
    title: 'Large Front-Mounted Carrying Handle',
  },
  {
    list: [{ text: 'A more compact design allows for easy transportation and storage.' }],
    title: 'Fold-Down Tiller Handle',
  },
  {
    list: [{ text: 'Standard 15" or 20" shaft lengths allow for a wider range of boat applications' }],
    title: 'Available In Two Shaft Lengths',
  },
];
