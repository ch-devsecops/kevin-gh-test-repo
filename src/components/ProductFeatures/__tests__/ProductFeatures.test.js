import React from 'react';
import { jssRender, cleanup, axe } from 'test-utils';
import ProductFeaturesContainer from '../ProductFeaturesContainer';
import featuresStrToArray from '../utils';
import { inputRequest, outputObject, outputObjectWithAccordion } from '../__mocks__/mockData';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/product-features/en.json');

describe('Test function featuresStrToArray', () => {
  test('transpiles TextFeatures string w/o Accordion', () => {
    const value = featuresStrToArray(inputRequest());
    expect(value).toEqual(outputObject);
  });
  test('transpiles TextFeatures string with Accordion', () => {
    const value = featuresStrToArray(inputRequest(true));
    expect(value).toEqual(outputObjectWithAccordion);
  });
});

describe('<ProductFeaturesContainer>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(<ProductFeaturesContainer features={mockData.fields.mappedFeat} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should return null if there is no fields', () => {
    const { container } = jssRender(<ProductFeaturesContainer features={null} />);

    expect(container).toMatchSnapshot();
  });

  test('should render even if there are empty fields', () => {
    const { container } = jssRender(<ProductFeaturesContainer features={[]} />);

    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(<ProductFeaturesContainer features={mockData.fields.mappedFeat} />);

    expect(await axe(container)).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
