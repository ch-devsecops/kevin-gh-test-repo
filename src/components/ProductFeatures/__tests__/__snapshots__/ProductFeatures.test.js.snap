// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductFeaturesContainer> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 deOXHd"
    data-testid="product-features-content"
    tabindex="0"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Fuel-Efficient, High-Output Operation
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Precision camshaft design offers precise valve timing and optimal valve overlap for better fuel efficiency
        </li>
        <li>
          OHV design for increased efficiency and optimal power transfer
        </li>
        <li>
          High compression ratio for better fuel efficiency
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Smooth Performance
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Precision-engineered components result in lower vibration
        </li>
        <li>
          Ball bearing supported crankshaft for greater stability
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Exceptionally Quiet
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Large capacity, multi-chamber exhaust system
        </li>
        <li>
          Reduced mechanical noise due to light weight, noise-reducing materials
        </li>
        <li>
          Forged steel crankshaft and rigid crankcase
        </li>
        <li>
          Sophisticated air intake system
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Proven Reliability
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Oil alert
          <div
            class="Box-sc-1bsf784-0 gTpLoe sc-1k44dfv-0 eVOXfr"
            height="14px"
            width="14px"
          >
            <button
              aria-label="show tooltip pop up"
              class="sc-1k44dfv-0 jOKBEK"
            />
          </div>
        </li>
        <li>
          Cast iron cylinder sleeve
        </li>
        <li>
          High-quality materials, fit, and finish
        </li>
        <li>
          Fuel Valve
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Easy to Use and Maintain
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Large fuel tanks
        </li>
        <li>
          Large automotive type fuel cap
        </li>
        <li>
          Easily accessible spark plug
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Easy Starting
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Heavy-duty recoil starter
        </li>
        <li>
          Automatic mechanical decompression system
          <div
            class="Box-sc-1bsf784-0 gTpLoe sc-1k44dfv-0 eVOXfr"
            height="14px"
            width="14px"
          >
            <button
              aria-label="show tooltip pop up"
              class="sc-1k44dfv-0 jOKBEK"
            />
          </div>
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Emissions Compliant
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          CARB and EPA certified
        </li>
        <li>
          No catalyst necessary
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Available Options
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Spark arrester available
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<ProductFeaturesContainer> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 deOXHd"
    data-testid="product-features-content"
    tabindex="0"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Fuel-Efficient, High-Output Operation
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Precision camshaft design offers precise valve timing and optimal valve overlap for better fuel efficiency
        </li>
        <li>
          OHV design for increased efficiency and optimal power transfer
        </li>
        <li>
          High compression ratio for better fuel efficiency
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Smooth Performance
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Precision-engineered components result in lower vibration
        </li>
        <li>
          Ball bearing supported crankshaft for greater stability
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Exceptionally Quiet
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Large capacity, multi-chamber exhaust system
        </li>
        <li>
          Reduced mechanical noise due to light weight, noise-reducing materials
        </li>
        <li>
          Forged steel crankshaft and rigid crankcase
        </li>
        <li>
          Sophisticated air intake system
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Proven Reliability
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Oil alert
          <div
            class="Box-sc-1bsf784-0 gTpLoe sc-1k44dfv-0 eVOXfr"
            height="14px"
            width="14px"
          >
            <button
              aria-label="show tooltip pop up"
              class="sc-1k44dfv-0 jOKBEK"
            />
          </div>
        </li>
        <li>
          Cast iron cylinder sleeve
        </li>
        <li>
          High-quality materials, fit, and finish
        </li>
        <li>
          Fuel Valve
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Easy to Use and Maintain
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Large fuel tanks
        </li>
        <li>
          Large automotive type fuel cap
        </li>
        <li>
          Easily accessible spark plug
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Easy Starting
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Heavy-duty recoil starter
        </li>
        <li>
          Automatic mechanical decompression system
          <div
            class="Box-sc-1bsf784-0 gTpLoe sc-1k44dfv-0 eVOXfr"
            height="14px"
            width="14px"
          >
            <button
              aria-label="show tooltip pop up"
              class="sc-1k44dfv-0 jOKBEK"
            />
          </div>
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Emissions Compliant
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          CARB and EPA certified
        </li>
        <li>
          No catalyst necessary
        </li>
      </ul>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 exTPCJ"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Available Options
      </h6>
      <ul
        class="sc-1k44dfv-0 sc-1k44dfv-0 iEOBXO"
      >
        <li>
          Spark arrester available
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<ProductFeaturesContainer> should render even if there are empty fields 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 deOXHd"
    data-testid="product-features-content"
    tabindex="0"
  />
</div>
`;

exports[`<ProductFeaturesContainer> should return null if there is no fields 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 deOXHd"
    data-testid="product-features-content"
    tabindex="0"
  />
</div>
`;
